crate/kmip/src/crypto/wrap/wrap_key.rs

```diff
@@ -187,17 +218,15 @@ pub(crate) fn wrap(
                         .and_then(|info| info.cryptographic_parameters)
                         .and_then(|params| params.block_cipher_mode);
                     let wrap_secret = key_block.key_bytes()?;
-                    let aad = additionnal_data_encryption.unwrap_or_default();
+                    let aad = additional_data_encryption.unwrap_or_default();
                     match block_cipher_mode {
                         Some(BlockCipherMode::GCM) => {
                             // wrap using aes Gcm
                             let aead = AeadCipher::Aes256Gcm;
                             let nonce = random_nonce(aead)?;
                             let (data, authenticated_encryption_tag) =
                                 aead_encrypt(aead, &wrap_secret, &nonce, aad, key_to_wrap)?;
-                            let ciphertext =
-                                [nonce.clone(), data, authenticated_encryption_tag.clone()]
-                                    .concat();
+                            let ciphertext = [nonce, data, authenticated_encryption_tag].concat();
```

Comment:
didn't you wanted to remove concat and use extend_from_slice function ?

Reply:
exact!